src/EditorFeatures/Core.Wpf/InlineRename/AbstractEditorInlineRenameService.SymbolRenameInfo.cs

```diff
@@ -241,6 +238,30 @@ public bool TryOnAfterGlobalSymbolRenamed(Workspace workspace, IEnumerable<Docum
                 return _refactorNotifyServices.TryOnAfterGlobalSymbolRenamed(workspace, changedDocumentIDs, RenameSymbol,
                     this.GetFinalSymbolName(replacementText), throwOnFailure: false);
             }
+
+            public InlineRenameFileRenameInfo GetFileRenameInfo()
+            {
+                var kindSupportsRename = RenameSymbol.Kind == SymbolKind.NamedType;
+                var allowRename = kindSupportsRename 
+                    && (RenameSymbol.Locations.Length == 1)
+                    && (OriginalNameMatches(_document, RenameSymbol.Name) || ContainsOnlyOneType(_document));
```

Comment:
the ContainsOnlyOneType seems a bit broad.  if i have a a file called `Helpers.cs` and i have a single type in it called `MyExtensions`, then should renaming `MyExtensions` cause the file to be renamed?  (maybe?  not sure?  i keep flip flopping :))

Reply:
I think that it's _fine_, because it's only _allowing_ the file to be renamed. If we want to fine tune the specifics of enabled by default or not I think that's reasonable as well. 

For the most part, we're actively suggesting you match file and type name. I don't see why the `Helpers` case would be different from a _suggestion_ point of view. Arguably the name still makes sense, but at least the system is behaving in a predictable way.